### PR TITLE
web: use stream search results for export

### PR DESCRIPTION
**Pull request by taras-yemets**
Closes https://github.com/sourcegraph/sourcegraph/issues/45718

Uses stream search results for export.

Previously we were using two different approaches for search results page and for search results export.
The search results page [uses stream search results](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@fc77a42710c2bc092abe6f504d0937508b2614f3/-/blob/client/web/src/search/results/StreamingSearchResults.tsx?L123) and the search export [called the GraphQL API](https://github.com/sourcegraph/sourcegraph-search-export) (this approach migrated  from [sourcegraph-search-export](https://github.com/sourcegraph/sourcegraph-search-export) extension). This caused inconsistencies in what users see on the results page and what results they download after clicking the 'Export results' button. 

This PR:
- refactors the search results export functionality to export the same results as displayed on the search results page
- adjusts the exported content to match the updated results structure ([`SearchMatch`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@fc77a42710c2bc092abe6f504d0937508b2614f3/-/blob/client/shared/src/search/stream.ts?L33) instead of [`SearchResultFields`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@fc77a42710c2bc092abe6f504d0937508b2614f3/-/blob/client/web/src/search/results/useExportSearchResultsQuery.ts?L31)): see [updated results table headers](https://github.com/sourcegraph/sourcegraph/pull/45702/files#diff-6c465a57e31d117928f2bfede9200f6b96fbd66449f6b6e972106116b089a5ceR22-R164)  

## Test plan
CI passes.

## App preview:

- [Web](https://sg-web-taras-yemets-fix-results-export.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/search/results/StreamingSearchResults.tsx [hunk 1]
```diff
@@ -375,7 +374,7 @@ export const StreamingSearchResults: FC<StreamingSearchResultsProps> = props =>
                         query={submittedURLQuery}
                         enableCodeInsights={codeInsightsEnabled && isCodeInsightsEnabled(props.settingsCascade)}
                         enableCodeMonitoring={enableCodeMonitoring}
-                        resultsFound={resultsFound}
```
Comment (philipp-spiess) [inline client/web/src/search/results/StreamingSearchResults.tsx:378]: Lol we had the results here already haha 😶‍🌫️

File: client/web/src/search/results/SearchActionsMenu.tsx [hunk 2]
```diff
@@ -21,36 +22,39 @@ import {
...
     createActions,
     createCodeMonitorAction,
     canCreateMonitor,
-    resultsFound,
     allExpanded,
     onExpandAllResultsToggle,
     onSaveQueryClick,
 }) => {
-    const [requestSearchResultsExport] = useExportSearchResultsQuery({ query, patternType, sourcegraphURL })
+    const resultsFound = useMemo<boolean>(() => (results ? results.results.length > 0 : false), [results])
```
Comment (philipp-spiess) [inline client/web/src/search/results/SearchActionsMenu.tsx:53]: I don't think you need to memo this, accessing `.length` and comparing a number is probably faster than the memo overhead

**Review by philipp-spiess — APPROVED** [review]
Niiceee!

File: client/web/src/search/results/SearchActionsMenu.tsx (hunk 2 above)
Comment (taras-yemets) [reply client/web/src/search/results/SearchActionsMenu.tsx]: Addressed in https://github.com/sourcegraph/sourcegraph/pull/45702/commits/1b1193debb653a2bfcaf673b798525d5623244ae

**Comment by philipp-spiess** [comment]
@taras-yemets I think we should add a change log entry as well, wdyt? I think in general I’m not adding that often enough 😓  but especially for customer reported issues it would be good to include it if they ever need to find out which version they have to update to.

**Comment by jasonhawkharris** [comment]
> @taras-yemets I think we should add a change log entry as well, wdyt? I think in general I’m not adding that often enough 😓  but especially for customer reported issues it would be good to include it if they ever need to find out which version they have to update to.

100% agree with a change log entry here. Otherwise LGTM! Thanks for getting it done so quickly!

**Review by jasonhawkharris — APPROVED** [review]

File: client/web/src/search/results/SearchActionsMenu.tsx [hunk 3]
```diff
@@ -21,36 +22,39 @@ import {
...
     allExpanded,
     onExpandAllResultsToggle,
     onSaveQueryClick,
 }) => {
-    const [requestSearchResultsExport] = useExportSearchResultsQuery({ query, patternType, sourcegraphURL })
+    const resultsFound = results ? results.results.length > 0 : false
+    const downloadResults = useCallback(
+        () => (results ? downloadSearchResults(results, sourcegraphURL, query) : undefined),
+        [results, sourcegraphURL, query]
+    )
```
Comment (keegancsmith) [inline client/web/src/search/results/SearchActionsMenu.tsx:57]: pinging a 3 month old PR, yay. But I think this is the root cause of a regression. There is a concept of display limit the streaming search API uses. Basically we only return 1500 results down to the browser (but continue streaming accurate result counts). Which means if you just use the results already downloaded to display you won't actually export all results. Here is were we hardcode that https://sourcegraph.com/github.com/sourcegraph/sourcegraph@db2922509bf00cd14fb30393a12d863974c15e88/-/blob/client/shared/src/search/stream.ts?L490

Here is the slack thread that just got started about this https://sourcegraph.slack.com/archives/CHEKCRWKV/p1678300443530819

Pinging here so hopefully one of the experts on this PR can turn around a fix :)

Comment (taras-yemets) [reply client/web/src/search/results/SearchActionsMenu.tsx:57]: Replied in the [referenced thread](https://sourcegraph.slack.com/archives/CHEKCRWKV/p1678313557765809?thread_ts=1678300443.530819&cid=CHEKCRWKV)

**Pull request merged**